tool/src/kotlin/snapshots/diplomat_tool__kotlin__test__struct.snap

```diff
@@ -67,7 +68,11 @@ internal class DiplomatCallback_MyNativeStruct_test_multi_arg_callback_diplomatC
                 }
             }
     @JvmField
-    internal var destructor: Pointer = Pointer(0L);
+    internal var destructor: Callback = object : Callback {
```

Comment:
Do you need to produce an extra `Callback` layer here? Is there any reason we can't just assign `destroy_rust_jvm_cookie` to this directly? Having two layers of indirection seems weird unless this is actually mandatory.

Reply:
I've tried this but it doesn't seem to work -- as I understand this, i'ts because the `Callback` is a JNA construct, and also you don't seem to be able to assign functions directly to variables in Kotlin :cry: 
But I'm not a Kotlin expert, maybe I'm missing something

Reply:
If we can figure out how to return a function pointer to the Rust function, from Rust, then we can probably have the destructor be a `Pointer` instead?